openprescribing/frontend/views/spending_utils.py

```diff
@@ -65,27 +53,25 @@ def ncso_spending_for_entity(entity, entity_type, num_months, current_month=None
 
 def ncso_spending_breakdown_for_entity(entity, entity_type, month):
     org_type, org_id = _get_org_type_and_id(entity, entity_type)
-    costs = _get_concession_cost_matrices(month, month, org_type, org_id)
-    # As we're only fetching costs for a single month we get matrices with just
-    # a single column, which we slice out below. It would be perfectly possible
-    # to generate a breakdown for a range of months, in which case we would use
-    # `numpy.sum(tariff_costs, axis=1)` to sum across dates. However, if we do
-    # this then we also need to alter the `_get_concession_prices` function to
-    # fetch tariff costs for all dates in the range, not just those dates on
-    # which there is a price concession.
-    tariff_costs = costs.tariff_costs[:, 0]
-    extra_costs = costs.extra_costs[:, 0]
-    quantities = costs.quantities[:, 0]
-    names = _get_names_for_bnf_codes(costs.bnf_code_offsets.keys())
+    quantities, tariff_costs, extra_costs = _get_concession_cost_matrices(
+        month, month, org_type, org_id
+    )
+    # We sum across columns (i.e. dates) to get the totals over time for each
+    # BNF code.
+    quantities = quantities.sum_columns()
```

Comment:
Would it be possible/sensible to make this something like `quantities.sum_over_dates()` (might require metaprogramming) or `quantities.sum('dates')`?

Reply:
Ah yes, good idea! I think I prefer the non-metaprogramming version, although the resulting API isn't quite as pretty.

Reply:
Or per my suggestion above

```python
quantities.aggregate(by="date", numpy.sum)
```